docs/rules/custom-rules.md

```diff
@@ -131,61 +162,97 @@ Notice we don't need to include `property` in this approach.
 ```yaml
 rules:
   assert/path-item-operation-required:
-    subject: PathItem
+    subject:
+      type: PathItem
+    assertions:
+      required:
+        - get
     message: Every path item must have a GET operation.
-    required:
-      - get
 ```
 
 The following example asserts that Tags have both name and description defined.
 
 ```yaml
 rules:
   assert/tag-name-and-desc-defined:
-    subject: Tag
-    property:
-      - name
-      - description
+    subject:
+      type: Tag
+      property:
+        - name
+        - description
+    assertions:
+      defined: true
     message: Every tag must have a name and description.
-    defined: true
 ```
 
 Another way to compose that rule is to require the subject keys:
 
 ```yaml
 rules:
   assert/tag-name-and-desc-required:
-    subject: Tag
-    message: Every tag must have a name and description.
-    required:
-      - name
-      - description
+    subject:
+      type: Tag
+    assertions:
+      required:
+        - name
+        - description
 ```
 
-### `context` example
+### `where` example
 
 The following example asserts that PUT responses with HTTP status 200 or 201 cannot return an `application/pdf`content type.
-Without the `context`, the assertion would evaluate every MediaTypesMap including:
-- Responses with all codes, including codes other than 200 or 201
+Without the `where`, the assertion would evaluate every `MediaTypesMap` property including:
+- Responses with all codes, including codes other than 200 or 201.
 - Responses for all HTTP methods, including DELETE, GET, POST, and more.
-To restrict the evaluation, use the `context` feature to limit what will be evaluated.
+To restrict the evaluation, use the `where` feature to limit what is evaluated.
 
 ```yaml
 assert/no-pdf-in-ok-response:
-  context:
-  - type: Operation
-    matchParentKeys: [put]
-  - type: Response
-    matchParentKeys: ['201', '200']
-  subject: MediaTypesMap
-  disallowed: ['application/pdf']
+  where:
+  - subject:
+      type: Operation
+      filterByParentKeys: 
+        - put
+    assertions:
+      defined: true
+  - subject:
+      type: Response
+      filterByParentKeys: 
+        - '201'
+        - '200'
+    assertions:
+      defined: true
+  subject:
+    type: MediaTypesMap
+  assertions:
+    disallowed:
+      - 'application/pdf'
+```
+
+Where enables complex assertions based on sibling values.
+The following example asserts that the `limit` parameter must have a schema with `type: integer`.
+
+```yaml
+assert/limit-is-integer:
+  subject:
+    type: Schema
+    property: type
+  assertions:
+    const: integer
+  where:
+    - subject:
+        type: Parameter
+        property: name
+      assertions:
+        const: limit
```

Comment:
Yes, that would be ideal:

```
    - subject:
        type: Parameter
        property: name
      assertions:
        const: limit
    - subject:
        type: Parameter
        property: in
      assertions:
        const: query
```

Reply:
I don't think it would be a quick fix. I'd left this as an improvement.

Reply:
I already have this problem and end up with need for creating custom plugin.

My case was to create assertion rule for specific names of query parameters. I wanted to pass defined list of possible names like: `q, offset, limit, sort` so they be consistent

Reply:
>What if we also want to apply this only to a Parameter which is located in: query for instance? 

Why. do not use an `enum` instead of `const`?

```
where:
  - subject:
      type: Parameter
      property: name
    assertions:
      enum:
        - limit
        - query
```

Reply:
I think @RomanHotsiy example is incomplete. Try this with the released beta.112:

```yaml
assert/query-parameter-names:
  where:
     - subject:
          type: Parameter
          property: in
        assertions:
          const: query
  subject:
     type: Parameter
     property: name
  assertions:
     enum:
        - q
        - offset
        - limit
        - sort
```